libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupBottomSheetDialogFragment.java

```diff
@@ -0,0 +1,100 @@
+package org.wordpress.android.login;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.DialogInterface.OnShowListener;
+import android.os.Bundle;
+import android.text.Html;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.material.bottomsheet.BottomSheetBehavior;
+
+import org.wordpress.android.login.widgets.WPBottomSheetDialogFragment;
+
+public class SignupBottomSheetDialogFragment extends WPBottomSheetDialogFragment {
+    public static final String TAG = SignupBottomSheetDialogFragment.class.getSimpleName();
+    private SignupSheetListener mSignupSheetListener;
+
+    public SignupBottomSheetDialogFragment(@NonNull final SignupSheetListener signupSheetListener) {
+        mSignupSheetListener = signupSheetListener;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        final View layout = inflater.inflate(R.layout.signup_bottom_sheet_dialog, container, false);
+
+        Button termsOfServiceText = layout.findViewById(R.id.signup_tos);
+        termsOfServiceText.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mSignupSheetListener.onSignupSheetTermsOfServiceClicked();
+            }
+        });
+        termsOfServiceText.setText(Html.fromHtml(String.format(
+                getDialog()
+                    .getContext()
+                    .getResources()
+                    .getString(R.string.signup_terms_of_service_text), "<u>", "</u>")));
+
+        Button signupWithEmailButton = layout.findViewById(R.id.signup_email);
+        signupWithEmailButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mSignupSheetListener.onSignupSheetEmailClicked();
+            }
+        });
+
+        Button signupWithGoogleButton = layout.findViewById(R.id.signup_google);
+        signupWithGoogleButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mSignupSheetListener.onSignupSheetGoogleClicked();
+            }
+        });
+
+        Dialog dialog = getDialog();
+        if (dialog != null) {
+            dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
+                @Override
+                public void onDismiss(DialogInterface dialog) {
+                    mSignupSheetListener.onSignupSheetCanceled();
```

Comment:
It seems this call is duplicated - we invoke the same callback in `public void onDismiss(DialogInterface dialog)`. Is this intentional?

Reply:
`dialog.setOnDismissListener` was initially added referring to [SimpleNote dialog](https://github.com/Automattic/simplenote-android/blob/6978afe6744db9c5f0a3141cccc1d3aa01c86772/Simplenote/src/main/java/com/automattic/simplenote/InfoBottomSheetDialog.java) code. It didn't seem to cover the scenario when dialog fragment was dismissed by touching on the screen which is why I added `public void onDismiss(DialogInterface dialog)` callback. I'll test it further and see if we can eliminate one of them.

Reply:
Fixed here d15dd92